### PR TITLE
First pass at a get_file() function

### PR DESCRIPTION
**Pull request by Zeitsperre**
## Overview

Changes:

* Added a function for collecting data files and caching them or, in the case of OPeNDAP links, opens and provides the data directly.

## Related Issue / Discussion

https://github.com/Ouranosinc/raven/issues/185

### CODE REVIEW

File: raven/tutorial.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    if suffix == "nc":
+        try:
+            ds = _open_dataset(local_file, **kwds)
+            if not cache:
+                ds = ds.load()
+                local_file.unlink()
+            return ds
+        except OSError:
+            raise
```
Comment (huard) [inline raven/tutorial.py:113]: I suggest `get_file` is only responsible for getting the file from upstream or the cache. Then could have another open_dataset() function that calls it and returns the corresponding `Dataset` object.

**Comment by Zeitsperre** [comment]
@huard this is good to be merged

**Review by huard — APPROVED** [review]
Nice !
How do you see the next step ? 
Is it to replace the links in `common.TESTDATA` by the urls ? Or use the links directly in the test files, without going through TESTDATA, or create pytest fixtures ?

**Comment by Zeitsperre** [comment]
> How do you see the next step ?
> Is it to replace the links in `common.TESTDATA` by the urls ? Or use the links directly in the test files, without going through TESTDATA, or create pytest fixtures ?

The `get_file` approach helps us not have to deal with URLs and adding them in would likely be more complicated (as access via URLs is managed already, and examining the cache should be the priority). I would probably just replace from `from xarray import open_dataset` to `from raven.tutorial  import open_dataset` in our tests. We don't need to do much.

**Pull request merged**